samples/csharp_dotnetcore/21.corebot-app-insights/Startup.cs

```diff
@@ -53,8 +50,31 @@ public void ConfigureServices(IServiceCollection services)
             // Create the Conversation state. (Used by the Dialog system itself.)
             services.AddSingleton<ConversationState>();
 
-            // The Dialog that will be run by the bot.
-            services.AddSingleton<MainDialog>();
+            // Register LUIS recognizer
+            services.AddSingleton<FlightBookingRecognizer>();
+
+            // Register the BookingDialog.
+            services.AddSingleton(sp =>
+            {
+                var telemetryClient = sp.GetService<IBotTelemetryClient>();
+                return new BookingDialog()
+                {
+                    TelemetryClient = telemetryClient
+                };
+            });
+
+            // The MainDialog that will be run by the bot passing and setting the TelemetryClient that will be used to log telemetry.
+            services.AddSingleton(sp =>
```

Comment:
Same as above. I personally prefer the DI way of injecting the IBotTelemetryClient (rather than the lamda)

Reply:
See above.